core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -307,6 +314,16 @@ private void doRemoveTask(TaskHandle taskHandle)
         log.debug("Task finished or failed %s", taskHandle.getTaskId());
     }
 
+    private synchronized void recordAllSplitQueueSize()
+    {
+        long now = ticker.read();
+        long timeDifference = now - this.lastSplitsQueueSizeRecordTime;
+        if (timeDifference > 0) {
+            this.allSplitsSizeDistribution.add(allSplits.size(), timeDifference);
```

Comment:
This should use previous queue length to update `allSplitsSizeDistribution`.

Reply:
Very good catch! :)